src/supa/connection/provider/server.py

```diff
@@ -396,8 +452,180 @@ def Provision(self, pb_provision_request: ProvisionRequest, context: ServicerCon
+                    )
                     provision_response = ProvisionResponse(header=to_response_header(pb_provision_request.header))
 
         log.debug("Sending response.", response_message=provision_response)
         return provision_response
+
+    def Release(self, pb_release_request: ReleaseRequest, context: ServicerContext) -> ReleaseResponse:
+        """Release reservation.
+
+        Check if the connection ID exists, if the provision state machine exists (as an indication
+        that the reservation was committed, and if the provision state machine transition is
+        allowed, all real work for releasing the reservation is done asynchronously by
+        :class:`~supa.job.reserve.ReleaseJob`
+
+        Args:
+            pb_release_request: Basically the connection id wrapped in a request like object
+            context: gRPC server context object.
+
+        Returns:
+            A response telling the caller we have received its release request.
+        """
+        connection_id = UUID(pb_release_request.connection_id)
+        log = logger.bind(method="Release", connection_id=str(connection_id))
+        log.debug("Received message.", request_message=pb_release_request)
+
+        from supa.db.session import db_session
+
+        with db_session() as session:
+            reservation = (
+                session.query(model.Reservation).filter(model.Reservation.connection_id == connection_id).one_or_none()
+            )
+            if reservation is None:
+                log.info("Connection ID does not exist")
+                release_response = ReleaseResponse(
+                    header=to_response_header(pb_release_request.header),
+                    service_exception=to_service_exception(
+                        NsiException(
+                            ReservationNonExistent, str(connection_id), {Variable.CONNECTION_ID: str(connection_id)}
+                        ),
+                        connection_id,
+                    ),
+                )
+            elif not reservation.provision_state:
+                log.info("First version of reservation not committed yet")
+                release_response = ReleaseResponse(
+                    header=to_response_header(pb_release_request.header),
+                    service_exception=to_service_exception(
+                        NsiException(
+                            InvalidTransition,
+                            "First version of reservation not committed yet",
+                            {
+                                Variable.CONNECTION_ID: str(connection_id),
+                                Variable.RESERVATION_STATE: reservation.reservation_state,
+                            },
+                        ),
+                        connection_id,
+                    ),
+                )
+            elif current_timestamp() > reservation.end_time:
+                log.info("Cannot release a reservation that is passed end time")
```

Comment:
Most of the log statements in these file could benefit from extra info.

`log.info("Cannot release a reservation that is passed end time", current_timestamp=current_timestamp(), reservation_end_time=reservation.end_time)`

Reply:
I see that most of the subclasses already setup a log.bind() : I didn't run it locally so it might already log enough. Feel very free to ignore. ;)